Dockerfile

```diff
@@ -19,14 +21,12 @@ RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone &
     apt-get clean && \
     rm -rf /var/lib/apt/lists/*
 
-RUN python3 -m pip install --user colorama ansiwrap attrs watchdog python-Levenshtein "mapfile-parser>=1.2.1,<2.0.0" "rabbitizer>=1.0.0,<2.0.0"
-RUN python3 -m pip install --upgrade attrs pycparser
+RUN python3 -m venv .venv && . .venv/bin/activate && \ 
+    python3 -m pip install colorama ansiwrap attrs watchdog python-Levenshtein "mapfile-parser>=1.2.1,<2.0.0" "rabbitizer>=1.0.0,<2.0.0" && \
+    python3 -m pip install --upgrade attrs pycparser
```

Comment:
From what I can tell, yes. Without it, I get an error. Apparently you are no longer supposed to use pip install without the virtual environment created first.

Sources:
- https://www.omgubuntu.co.uk/2023/04/pip-install-error-externally-managed-environment-fix
- https://stackoverflow.com/questions/75608323/how-do-i-solve-error-externally-managed-environment-every-time-i-use-pip-3

Reply:
I mean, can we remove lines 24-26 entirely? They will be installed in the virtual environment anyway during `make setup` (https://github.com/zeldaret/oot/blob/b270d58f4386c1f76993d82123b78dd553f44f0a/Makefile#L652) so it would be nice if we didn't have to worry about it here.

Reply:
Oh, nevermind. I think I see what you mean. One second.

Reply:
There are some differing requirements here between the two. Let me see what I can do.

Reply:
I think `requirements.txt` is up-to-date. I believe extras here are `ansiwrap` and `attrs`, which asm-differ/diff.py once depended on, but neither of these are required anymore:

https://github.com/simonlindholm/asm-differ/commit/8f81e1024e55aba6a73d0e12acb1bc50519345f5
https://github.com/simonlindholm/asm-differ/commit/79fcecfe59667c3ada5e281b3fd2216b47615d3b